pinc/theme.inc

```diff
@@ -270,43 +369,52 @@ function html_navbar()
         // on the same line.
         echo "<span id='navbar-right'>";
         $links = [];
-        if (get_url_for_inbox()) {
-            $inbox_text = _("Inbox");
-
-            $numofPMs = get_number_of_unread_messages($user->username);
-            if ($numofPMs > 0) {
-                $inbox_text .= " " . sprintf(_("(%s unread)"), $numofPMs);
-            }
-            $links[] = ['text' => $inbox_text, 'url' => get_url_for_inbox()];
-        }
+        $links[] = ['text' => _("Stats"), 'url' => "stats/stats_central.php"];
 
-        if (get_url_for_forum()) {
-            $links[] = ['text' => _("Forums"), 'url' => get_url_for_forum()];
-        }
+        echo headerbar_text_array($links);
+        echo "  &#183; ";
 
-        if (!empty($blog_url)) {
-            $links[] = ['text' => _("Blog"), 'url' => $blog_url];
-        }
+        $account_links = [];
+        $account_links[] = ['text' => _("Profile"), 'url' => "stats/members/mdetail.php"];
+        $account_links[] = ['text' => _("Preferences"), 'url' => "userprefs.php"];
+        $account_links[] = ['text' => _("Log Out"), 'url' => "accounts/logout.php"];
 
-        if (!empty($wiki_url)) {
-            $links[] = ['text' => _("Wiki"), 'url' => $wiki_url];
-        }
+        echo headerbar_menu($user->username, $account_links, true);
 
-        $links[] = ['text' => _("Stats"), 'url' => "stats/stats_central.php"];
-        $links[] = ['text' => _("Prefs"), 'url' => "userprefs.php"];
-        $links[] = ['text' => _('Help'), 'url' => get_faq_url("faq_central.php")];
-        $links[] = ['text' => sprintf(_("Log Out (%s)"), $user->username), 'url' => "accounts/logout.php"];
-
-        echo headerbar_text_array($links);
         echo "</span>";
     }
 
     echo "</div>\n";
     echo "</div>\n";
 }
 
+function headerbar_menu(string $menu_title, array $links, bool $align_right = false): string
+{
+    $menu_style = "";
+    $menu_contents_style = "";
+    if ($align_right) {
+        $menu_style = "style='padding-right: 0.5em;'";
+        $menu_contents_style = "style='right: 0;'";
+    }
+
+    $menu_id = 'menu' . md5($menu_title);
+    $output = [];
```

Comment:
Again, perhaps single assign here rather than repeated append. Makes it a little less noisy.

Reply:
Yup, done.